modules/codegen/src/main/scala/com/twilio/guardrail/extract/package.scala

```diff
@@ -1,19 +1,32 @@
 package com.twilio.guardrail
 
 package object extract {
-  def ScalaTracingLabel[F: VendorExtension.VendorExtensible](v: F): Option[String] =
-    VendorExtension(v).extract[String]("x-scala-tracing-label")
-  def ScalaPackage[F: VendorExtension.VendorExtensible](v: F): Option[String] =
-    VendorExtension(v).extract[String]("x-scala-package")
-  def ScalaType[F: VendorExtension.VendorExtensible](v: F): Option[String] =
-    VendorExtension(v).extract[String]("x-scala-type")
+  private def extractWithFallback[F: VendorExtension.VendorExtensible, T: Extractable](v: F, name: String, fallbackName: String): Option[T] = {
+    val ve = VendorExtension(v)
+    ve.extract[T](name)
+      .orElse({
+        val value = ve.extract[T](fallbackName)
+        if (value.isDefined) {
+          println(s"WARNING: Deprecated vendor extension '${fallbackName}'; please migrate to use '${name}' instead")
+        }
+        value
+      })
+  }
+
+  def TracingLabel[F: VendorExtension.VendorExtensible](v: F): Option[String] =
+    extractWithFallback[F, String](v, "x-tracing-label", "x-scala-tracing-label")
+
+  def JvmPackage[F: VendorExtension.VendorExtensible](v: F): Option[String] =
+    extractWithFallback[F, String](v, "x-jvm-package", "x-scala-package")
```

Comment:
Thoughts on having `x-scala-package`, `x-java-package`, and `x-jvm-package`? Different aesthetics for organization by language, perhaps? Alternately, diamond dependencies on one client generated for Scala and the same one generated for Java?

Reply:
Ehhhhhhhhh, does this really seem worth pre-emptively adding before someone asks for it?

Reply:
I wouldn't think twice about it other than "This is deprecated! Change to this other thing!" _later_ "We brought it back!"

Reply:
Yeah, true.  If you really want it, I'll add it; just annoying because it means adding another ScalaTerm a la `CustomTypePrefixes`

Reply:
I can't honestly say I'm comfortable merging without it. This might be an opportunity to generalize the previous one-off target-interpolating setting hierarchy?

Reply:
Sure.  Got a problem, though.  `JvmPackage` is used in `SwaggerGenerator` (in `GetClassName`), which is conceptually lower in the stack than `ScalaGenerator`/`JavaGenerator` (which is what provides the list of language-specific prefixes to use).

One possibility I can think of is to put a call to ScalaTerm's `customPrefixes()` in `Common.prepareDefinitions()` (which is the only place that calls `getClassName()`, and also already has a reference to `ScalaTerms`), and then add a param to `getClassName()` to pass those prefixes in, so it can do the vendor extension extraction using the prefix list.  (Looks like `ServerGenerator` also calls `getClassName()`, so that call site would have to be changed as well.)

What do you think?

Reply:
(Ended up pushing a couple more commits to do just that.  LMK if that seems reasonable.)